plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -4862,10 +4862,15 @@ public synchronized Network setupvSwitchNetwork(final Connection conn) {
         return null;
     }
 
-    public void shutdownVM(final Connection conn, final VM vm, final String vmName) throws XmlRpcException {
+    public void shutdownVM(final Connection conn, final VM vm, final String vmName, final boolean forcedStop) throws XmlRpcException {
         Task task = null;
         try {
-            task = vm.cleanShutdownAsync(conn);
+            if (forcedStop) {
```

Comment:
@koushik-das the stopVM command will have the forced-flag, using that one can investigate whether a clean shutdown was attempted or a forced/hard shutdown.